cli/compose/template/template.go

```diff
@@ -32,8 +32,20 @@ func (e InvalidTemplateError) Error() string {
 // and the absence of a value.
 type Mapping func(string) (string, bool)
 
-// Substitute variables in the string with their values
-func Substitute(template string, mapping Mapping) (string, error) {
+// SubstituteFunc is a user-supplied function that apply substitution.
+// Returns the value as a string, a bool indicating if the function could apply
+// the substitution and an error.
+type SubstituteFunc func(string, Mapping) (string, bool, error)
+
+// SubstituteWith subsitute variables in the string with their values.
+// It accepts additionnal substitute function.
+func SubstituteWith(template string, mapping Mapping, pattern *regexp.Regexp, subsFuncs ...SubstituteFunc) (string, error) {
+	subsFuncs = append([]SubstituteFunc{
```

Comment:
Should we always _append_ the default functions, or only do so if no custom functions are provided?

Reply:
Good point. I did that to not *export* the default function, but looking back at it, this function should probably just take the list and use them (letting user decide which one to apply)